app/controllers/profile_pins_controller.rb

```diff
@@ -0,0 +1,30 @@
+class ProfilePinsController < ApplicationController
+  before_action :authenticate_user!, only: %i[create]
+
+  def create
+    @profile_pin = ProfilePin.new
+    @profile_pin.profile_id = current_user.id
+    @profile_pin.profile_type = "User"
+    @profile_pin.pinnable_id = profile_pin_params[:pinnable_id].to_i
+    @profile_pin.pinnable_type = "Article"
+    if @profile_pin.save!
+      flash[:pins_success] = "📌 Pinned! (pinned posts display chronologically, 5 max)"
+    else
+      flash[:pins_error] = "You can only have five pins"
+    end
+    redirect_back(fallback_location: "/dashboard")
+  end
+
+  def update
```

Comment:
Is there are reason why this is not the destroy action?

Reply:
I'd need to look further into whether we have addressed this in updates since the fact, but going way back to this project's creation I removed the jquery dependency for Rails that was actually required for calling `destroy` actions. I know Rails has since removed jquery as a dependency from the `ujs` approach but I believe it is still the case that we can't call destroy actions in the Rails way without importing the specific dependencies, which we do on the "are you sure you want to destroy this" page.

We should definitely make get this documented and make some of these overall concepts around how we cache pages more clearly documented in our Readme/docs.

Anyway, I figured in and of itself `update` was close enough to what's going on with updating the state of something's pinned status that it's clean enough for now.

Reply:
@benhalpern there's an example in the code that does something similar I guess. If you look at the code for removing the API key:

https://github.com/thepracticaldev/dev.to/blob/dcbb188cf1e3ddbbf87711a43837766eb609676b/app/views/users/_account.html.erb#L47-L51

that's a POST with the `method: :delete` trick that ends up in the `destroy` action inside the api secrets controller https://github.com/thepracticaldev/dev.to/blob/master/app/controllers/api_secrets_controller.rb#L20